mico-core/src/main/java/io/github/ust/mico/core/service/GitHubCrawler.java

```diff
@@ -53,48 +61,115 @@ public GitHubCrawler(RestTemplate restTemplate, KubernetesNameNormalizer kuberne
+        }
+
+        if (responseBasicInfo.getStatusCode().is2xxSuccessful() && responseReleaseInfo.getStatusCode().is2xxSuccessful()) {
+            ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
             JsonNode basicInfoJson = mapper.readTree(responseBasicInfo.getBody());
             JsonNode releaseInfoJson = mapper.readTree(responseReleaseInfo.getBody());
 
             String name = basicInfoJson.get("name").textValue();
             String normalizedName = kubernetesNameNormalizer.normalizeName(name);
 
-            return new MicoService()
+            MicoService micoService = new MicoService()
                 .setShortName(normalizedName)
                 .setName(basicInfoJson.get("full_name").textValue())
                 .setVersion(releaseInfoJson.get("tag_name").textValue())
                 .setDescription(basicInfoJson.get("description").textValue())
                 .setServiceCrawlingOrigin(MicoServiceCrawlingOrigin.GITHUB)
                 .setGitCloneUrl(basicInfoJson.get("clone_url").textValue())
                 .setGitReleaseInfoUrl(releaseInfoJson.get("url").textValue());
+            if (!StringUtils.isEmpty(dockerfilePath)) {
+                if (existsFileInGithubRepo(gitHubRepositoryApiUrl, dockerfilePath)) {
+                    micoService.setDockerfilePath(dockerfilePath);
+                } else {
+                    throw new IllegalArgumentException("The Dockerfile path must be a valid path relativ to the repository root");
+                }
+            }
+            return micoService;
+        } else {
+            throw new IllegalArgumentException("An error occurred while requesting information about the GitHub repository. Send two requests and got the status codes "
+                + responseBasicInfo.getStatusCode() + " and " + responseReleaseInfo.getStatusCode().is2xxSuccessful());
+        }
 
-        } catch (IOException e) {
-            log.error(e.getStackTrace().toString());
-            log.error("Getting exception '{}'", e.getMessage());
-            throw e;
+    }
+
+    /**
+     * Checks if a file in a GitHub repository exists
+     *
+     * @param uriBasicInfo   the URI to a specific release of a GitHub repository
+     * @param dockerfilePath the relative path to the Dockerfile. The path is relative to the root directory of the
+     *                       GitHub repository
+     * @return {@code true} if the Dockerfile exists in the in the specified repository. {@code false} if response status is HTTP NOT FOUND.
+     * @throws {@link HttpClientErrorException} in case of HTTP 4XX response from GitHub
+     * @throws {@link HttpStatusCodeException} in case of HTTP 5XX response from GitHub
+     */
+    private boolean existsFileInGithubRepo(String uriBasicInfo, String dockerfilePath) {
+        ResponseEntity<String> responseDockerFileInfo;
```

Comment:
To be consistent, the name should be `responseDockerfileInfo`

Reply:
Done